### PR TITLE
Bugfix: Tar files created in their target directory shouldn't recurse infinitely

### PR DESCRIPTION
**Pull request by drewstinnett**
This addresses #383 .

Happy to refactor however you like, and open to any feedback. Thank you!

### CODE REVIEW

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for the patch! I have not encountered this situation so I appreciate the test case. Is the right behavior really to just skip the file?

**Comment by drewstinnett** [comment]
> Thanks for the patch! I have not encountered this situation so I appreciate the test case. Is the right behavior really to just skip the file?

Yep! At least to the best of my knowledge skipping this sub '.' entry keeps everything going, correctly. I tested it on some random archives here and they all behaved correctly with that weird entry being completely skipped.

File: fs.go [hunk 1]
```diff
@@ -567,11 +567,15 @@ func (f ArchiveFS) ReadDir(name string) ([]fs.DirEntry, error) {
...
 		file.NameInArchive = strings.Trim(file.NameInArchive, "/")
+		if file.NameInArchive == "." {
+			return nil
+		}
 		files = append(files, file)
 		if file.NameInArchive == name && !file.IsDir() {
 			foundFile = true
 			return errStopWalk
 		}
+		}
```
Comment (mholt) [inline fs.go:578]:
```suggestion
```

Oops, my bad

**Review by mholt — APPROVED** [review]
LGTM after I fix my syntax error

**Pull request merged**